### PR TITLE
fix smallBalances calculation

### PR DESCRIPTION
**Pull request by skylarbarrera**
Noticed the "smallBalance" AKA extra tokens section's sum was off. Fixed the logic so that is divides assets & calculates properly. 

before: https://cloud.skylarbarrera.com/IMG_0568.jpeg

after: https://cloud.skylarbarrera.com/Screen-Shot-2020-10-19-at-9.36.09-PM.png

### CODE REVIEW

File: src/helpers/assets.js [hunk 1]
```diff
@@ -47,7 +46,7 @@ export const buildCoinsList = (
 
   const isShortList = assetsLength <= amountOfShowedCoins;
 
-  forEach(assets, asset => {
+  forEach(assets, function(asset, index) {
```
Comment (brunobar79) [inline src/helpers/assets.js:49]: you don't need to add `function`, just keep the arrow and add parenthesis 

` forEach(assets, (asset, index) => {`

**Review by brunobar79 — COMMENTED** [review]
Code looks good to me.

However I think there's another bug because we should be including Pools as part of the total wallet balance, considering that Pools is a subsection of balances with the current UI hierarchy.

@christianbaroni  WDYT?

**Comment by christianbaroni** [comment]
> Code looks good to me.
> 
> However I think there's another bug because we should be including Pools as part of the total wallet balance, considering that Pools is a subsection of balances with the current UI hierarchy.
> 
> @christianbaroni WDYT?

agreed we should include in total

**Comment by brunobar79** [comment]
@skylarbarrera wanna handle this on this PR whenever you have a chance?

**Review by jinchung — CHANGES_REQUESTED** [review]
Just wanted to drop a note here before merging for changes related to the uniswap total number as well as the showcasing logic! Will follow up separately over slack

**Comment by jinchung** [comment]
Yo for reference and FYI:

The assets object is ordered by $, not by hidden vs not etc.  So we unfortunately cannot do the small vs large balances check solely based on the index of an asset.

Examples:
index 0 - ETH (pinned)
index 1 - DAI (not pinned)
index 2 - YFI (not pinned)
index 3 - Loopring (not pinned)
index 4 - Synthetix (not pinned)
This should show all the items above, but at index 4, we would do 4 < 5 - 1 (amountOfShowedCoins - pinned coins), and 4 is not < 4 therefore we would put it incorrectly in the small balances list.

index 0 - ETH (pinned)
index 1 - DAI (hidden)
index 2 - YFI (not pinned)
index 3 - Loopring (not pinned)
index 4 - Synthetix (not pinned)
index 5 - something else (not pinned)
so in this case, at index 5, we should be showing it bc DAI is hidden, but bc it has index 5, and it is not < amountOfShowedCoins, then we say it is part of the small balances. The hidden DAI which as the beginning of the list bc of $ sort, is messing up our usage of the index.

File: src/hoc/uniswapLiquidityTokenInfoSelector.js [hunk 2]
```diff
@@ -100,7 +100,7 @@ const buildUniswapCards = (
 
   return {
     uniswap: orderedUniswapPools,
-    uniswapTotal: convertAmountToNativeDisplay(uniswapTotal, nativeCurrency),
+    uniswapTotal: uniswapTotal,
```
Comment (jinchung) [inline src/hoc/uniswapLiquidityTokenInfoSelector.js:103]:
```suggestion
    uniswapTotal,
```

File: src/helpers/buildWalletSections.js [hunk 3]
```diff
@@ -90,15 +90,18 @@ const addEth = section => {
...
       isCoin: true,
       isPinned: true,
+      isPlaceholder: true,
       isSmall: false,
       name: 'Ethereum',
       native: {
         balance: {
           amount: '0',
-          display: '$0',
+          display: '$0.00',
```
Comment (jinchung) [inline src/helpers/buildWalletSections.js:104]: hardcoded $

File: src/components/send/SendAssetList.js [hunk 4]
```diff
@@ -80,9 +80,18 @@ export default class SendAssetList extends React.Component {
       assets.pop(); // removes not needed coin divider
     }
 
+    //check for placeholder ETH & remove
+    assets[0]?.isPlaceholder ? assets.shift() : null;
     const visibleAssetsLength = assets.length;
+
     this.data = assets;
+    console.log(this.data);
```
Comment (jinchung) [inline src/components/send/SendAssetList.js:88]: can remove these console logs

File: src/components/send/SendAssetList.js [hunk 5]
```diff
@@ -80,9 +80,18 @@ export default class SendAssetList extends React.Component {
...
+    console.log(this.data.length);
     if (smallBalances.assets.length > 0) {
+      //check for placeholder ETH & remove
+      smallBalances.asset = smallBalances.assets.filter(
```
Comment (jinchung) [inline src/components/send/SendAssetList.js:92]: did you mean smallBalances.assets on the left hand side? (not smallBalances.asset)

File: src/components/send/SendAssetList.js [hunk 6]
```diff
@@ -80,9 +80,18 @@ export default class SendAssetList extends React.Component {
...
+        asset => asset?.isPlaceholder
```
Comment (jinchung) [inline src/components/send/SendAssetList.js:93]: did you mean !asset?.isPlaceholder (keeping anything that is not a placeholder)

File: src/components/send/SendAssetList.js [hunk 7]
```diff
@@ -80,9 +80,18 @@ export default class SendAssetList extends React.Component {
       assets.pop(); // removes not needed coin divider
     }
 
+    //check for placeholder ETH & remove
+    assets[0]?.isPlaceholder ? assets.shift() : null;
```
Comment (jinchung) [inline src/components/send/SendAssetList.js:84]: regarding index 0: what happens in the scenario if ETH is pinned but there are other assets that are pinned with larger balances (and come in before it?)

File: src/components/expanded-state/ChartExpandedState.js [hunk 8]
```diff
@@ -132,6 +134,9 @@ export default function ChartExpandedState({ asset }) {
     asset.uniqueId,
   ]);
 
+  const needsEth =
+    asset.address === 'eth' && asset.balance.amount === '0' ? true : false;
```
Comment (jinchung) [inline src/components/expanded-state/ChartExpandedState.js:138]: don't need the ternary here ( ? true : false)

File: src/components/send/SendAssetList.js (hunk 7 above)
Comment (jinchung) [inline src/components/send/SendAssetList.js:84]: regarding index 0: what happens in the scenario if ETH is pinned/placeholder but there are other assets that are pinned with larger balances (and come in before it?)

File: src/components/send/SendAssetList.js (hunk 6 above)
Comment (skylarbarrera) [reply src/components/send/SendAssetList.js:93]: i did, it was working anyways because of the ?

File: src/components/send/SendAssetList.js (hunk 7 above)
Comment (skylarbarrera) [reply src/components/send/SendAssetList.js:84]: I added isPlaceholder, it only exists for the hardcoded ETH object(s)

Comment (jinchung) [reply src/components/send/SendAssetList.js:84]: what if ETH is not the first item in the assets list?

**Comment by jinchung** [comment]
If you do: unpin ETH, then kill the app and restart, ETH is pinned again

File: src/redux/editOptions.js [hunk 9]
```diff
@@ -179,7 +179,7 @@ const INITIAL_STATE = {
   currentAction: EditOptions.none,
   hiddenCoins: [],
   isCoinListEdited: false,
-  pinnedCoins: [],
+  pinnedCoins: ['eth'],
```
Comment (jinchung) [inline src/redux/editOptions.js:182]: We may need to do the isPinned / pinnedCoins list a bit differently to properly handle when a user has explicitly unpinned ETH

Comment (skylarbarrera) [reply src/redux/editOptions.js:180]: this was a preexisting issue via https://github.com/rainbow-me/rainbow/blob/6dce737518e6bfc448742ac2b0e5ece91a431cc7/src/redux/editOptions.js#L28

removed it and fixed!

**Review by jinchung — APPROVED** [review]

File: src/components/expanded-state/ChartExpandedState.js [hunk 10]
```diff
@@ -201,29 +205,35 @@ export default function ChartExpandedState({ asset }) {
...
+                style: { flex: 1 },
+              }}
+            />
+          )}
+          <SendActionButton
             color={color}
-            inputType={AssetInputTypes.in}
             radiusAndroid={24}
-            radiusWrapperStyle={{ flex: 1, marginRight: 10 }}
+            radiusWrapperStyle={{ flex: 1, marginLeft: 10 }}
```
Comment (brunobar79) [inline src/components/expanded-state/ChartExpandedState.js:229]: I've changed this so many times the past week that I'm forced to ask I if it's really needed and tested on both OS.

File: src/helpers/assets.js [hunk 11]
```diff
@@ -26,27 +29,90 @@ export const buildAssetUniqueIdentifier = item => {
...
+
+    const zeroEth = {
+      address: 'eth',
+      balance: {
+        amount: '0',
+        display: '0 ETH',
+      },
+      color: '#29292E',
+      decimals: 18,
+      icon_url: 'https://s3.amazonaws.com/token-icons/eth.png',
```
Comment (brunobar79) [inline src/helpers/assets.js:52]: This is being used in a few places. Should be moved to a const somewhere else

**Review by brunobar79 — APPROVED** [review]
💯

**Pull request merged**